src/core/use_cases/ark_url_info_processor.rs

```diff
@@ -0,0 +1,444 @@
+        self.generate_resource_iri(ark_info)
+    }
+
+    fn generate_dsp_redirect_url(&self, ark_info: &ArkUrlInfo) -> ArkUrlInfoResult<String> {
+        self.generate_dsp_redirect_url(ark_info)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::core::errors::ark_url_info::ArkUrlInfoError;
+
+    // Mock implementations for testing
+    struct MockParser;
+    struct MockConfig;
+    struct MockTemplate;
+    struct MockUuidGenerator;
+
+    impl ArkUrlParsingPort for MockParser {
+        fn parse_ark_v1(&self, _ark_id: &str) -> Option<(u32, Option<String>, Option<String>, Option<String>, Option<String>)> {
+            Some((1, Some("0001".to_string()), Some("resource123".to_string()), None, None))
+        }
+
+        fn parse_ark_v0(&self, _ark_id: &str) -> Option<(String, String, Option<String>)> {
+            Some(("0001".to_string(), "resource123".to_string(), None))
+        }
+
+        fn unescape_and_validate_uuid(&self, _ark_url: &str, escaped_uuid: &str) -> ArkUrlInfoResult<String> {
+            Ok(escaped_uuid.to_string())
+        }
+    }
+
+    impl ConfigurationPort for MockConfig {
+        fn get_dsp_ark_version(&self) -> u32 {
+            1
+        }
+
+        fn is_version_0_allowed(&self, _project_id: &str) -> ArkUrlInfoResult<bool> {
+            Ok(true)
+        }
+
+        fn get_top_level_redirect_url(&self) -> String {
+            "https://example.com/top".to_string()
+        }
+
+        fn get_project_template(&self, _project_id: &str, _template_name: &str) -> ArkUrlInfoResult<String> {
+            Ok("https://example.com/template".to_string())
+        }
+
+        fn get_project_host(&self, _project_id: &str) -> ArkUrlInfoResult<String> {
+            Ok("example.com".to_string())
+        }
+    }
+
+    impl TemplatePort for MockTemplate {
+        fn substitute(&self, _template: &str, _values: &HashMap<String, String>) -> ArkUrlInfoResult<String> {
+            Ok("https://example.com/substituted".to_string())
+        }
+
+        fn url_encode(&self, input: &str) -> ArkUrlInfoResult<String> {
+            Ok(input.to_string())
+        }
+    }
+
+    impl UuidGenerationPort for MockUuidGenerator {
+        fn generate_v5_uuid(&self, _input: &str) -> ArkUrlInfoResult<String> {
+            Ok("generated-uuid".to_string())
+        }
+    }
+
+    #[test]
+    fn test_parse_ark_v1_success() {
+        let processor = ArkUrlInfoProcessor::new(
+            MockParser,
+            MockConfig,
+            MockTemplate,
+            MockUuidGenerator,
+        );
+
+        let result = processor.parse_ark_id("ark:/12345/1/0001/resource123");
+        assert!(result.is_ok());
+        
+        let ark_info = result.unwrap();
+        assert_eq!(ark_info.url_version, 1);
+        assert_eq!(ark_info.project_id, Some("0001".to_string()));
+        assert_eq!(ark_info.resource_id, Some("resource123".to_string()));
+    }
```

Comment:
Some of these tests, I'm not sure how much they actually verify, apart from the behaviour of the mock.